### PR TITLE
Sanitize reason messages until we have them all updated

### PR DESCRIPTION
**Pull request by tmjd**
## For PR author

- [ ] Tests for change.
- [ ] If changing pkg/apis/, run `make gen-files`
- [ ] If changing versions, run `make gen-versions`

## For PR reviewers

A note for code reviewers - all pull requests must have the following:

- [ ] Milestone set according to targeted release.
- [ ] Appropriate labels:
  - `kind/bug` if this is a bugfix.
  - `kind/enhancement` if this is a a new feature.
  - `enterprise` if this PR applies to Calico Enterprise only.

### CODE REVIEW

File: pkg/controller/status/status.go [hunk 1]
```diff
@@ -79,6 +80,16 @@ type StatusManager interface {
 	SetMetaData(meta *metav1.ObjectMeta)
 }
 
+// This regex matches any characters that are not allowed in a Status Reason
+var reasonInvalidCharacters = regexp.MustCompile(`[^A-Za-z0-9_,:]`)
+
+// This regex matches any characters not allows at the start of a Status Reason
```
Comment (rene-dekker) [inline pkg/controller/status/status.go:86]: nit:
"allows" -> "allowed" . Happens 2x.
Also, it's best to end the comments with a period.

File: pkg/controller/status/status.go [hunk 2]
```diff
@@ -79,6 +80,16 @@ type StatusManager interface {
...
+var reasonInvalidStartCharacters = regexp.MustCompile(`^[^A-Za-z]*`)
```
Comment (rene-dekker) [inline pkg/controller/status/status.go:87]: Why is this plural again for the start character and why does it have the asterisk?

Comment (rene-dekker) [reply pkg/controller/status/status.go:87]: Never mind.

Comment (tmjd) [reply pkg/controller/status/status.go:87]: I had to think about that a bit myself when I looked at this. I'll add more to the comments to try to clarify what it is used for.

**Review by rene-dekker — APPROVED** [review]
lgtm

**Comment by tmjd** [comment]
I received verbal approval from @rene-dekker that these changes still look good.

**Pull request merged**